From 72e4d6b72841869573c9f170b145ea828a729069 Mon Sep 17 00:00:00 2001
From: Shammah Chancellor <shammah.chancellor@notvalid.com>
Date: Mon, 19 Oct 2026 06:46:15 -0700
Subject: [PATCH 19/28] test: declare the p2p tests' parameters as class
 attributes

Applies contrib/devtools/declarative-test-params.py to the p2p_* tests:
44 of 45 converted. p2p_headers_sync_with_minchainwork.py computes its
parameters and keeps set_test_params().

Every converted module imports, and the full functional suite gives the
same per-test results as before.
---
 test/functional/p2p_1p1c_network.py           | 11 +++----
 test/functional/p2p_add_connections.py        | 17 +++++-----
 test/functional/p2p_addr_relay.py             |  5 ++-
 test/functional/p2p_addrfetch.py              |  5 ++-
 test/functional/p2p_addrv2_relay.py           |  7 ++---
 test/functional/p2p_block_sync.py             |  5 ++-
 test/functional/p2p_blockfilters.py           | 15 +++++----
 test/functional/p2p_blocksonly.py             |  5 ++-
 test/functional/p2p_compactblocks.py          |  9 +++---
 .../p2p_compactblocks_blocksonly.py           |  5 ++-
 test/functional/p2p_compactblocks_hb.py       |  5 ++-
 test/functional/p2p_disconnect_ban.py         |  5 ++-
 test/functional/p2p_dos_header_tree.py        |  9 +++---
 test/functional/p2p_eviction.py               | 31 +++++++++----------
 test/functional/p2p_feefilter.py              | 27 ++++++++--------
 test/functional/p2p_filter.py                 |  7 ++---
 test/functional/p2p_fingerprint.py            |  5 ++-
 test/functional/p2p_getaddr_caching.py        |  9 +++---
 test/functional/p2p_getdata.py                |  3 +-
 test/functional/p2p_i2p_ports.py              |  7 ++---
 test/functional/p2p_i2p_sessions.py           | 13 ++++----
 test/functional/p2p_ibd_stalling.py           |  5 ++-
 test/functional/p2p_ibd_txrelay.py            | 13 ++++----
 test/functional/p2p_initial_headers_sync.py   |  5 ++-
 test/functional/p2p_inv_download.py           | 17 +++++-----
 test/functional/p2p_invalid_block.py          |  7 ++---
 test/functional/p2p_invalid_locator.py        |  3 +-
 test/functional/p2p_invalid_messages.py       |  7 ++---
 test/functional/p2p_invalid_tx.py             | 13 ++++----
 test/functional/p2p_leak.py                   |  3 +-
 test/functional/p2p_leak_tx.py                | 25 +++++++--------
 test/functional/p2p_message_capture.py        |  7 ++---
 test/functional/p2p_mutated_blocks.py         |  5 ++-
 test/functional/p2p_net_deadlock.py           |  3 +-
 test/functional/p2p_nobloomfilter_messages.py |  7 ++---
 test/functional/p2p_node_network_limited.py   |  7 ++---
 test/functional/p2p_opportunistic_1p1c.py     | 17 +++++-----
 test/functional/p2p_orphan_handling.py        |  5 ++-
 test/functional/p2p_permissions.py            |  5 ++-
 test/functional/p2p_ping.py                   | 11 +++----
 test/functional/p2p_sendheaders.py            |  7 ++---
 test/functional/p2p_timeouts.py               |  9 +++---
 test/functional/p2p_tx_privacy.py             |  3 +-
 test/functional/p2p_unrequested_blocks.py     |  7 ++---
 44 files changed, 176 insertions(+), 220 deletions(-)

diff --git a/test/functional/p2p_1p1c_network.py b/test/functional/p2p_1p1c_network.py
index 0c993d80fe..0adfaf5910 100644
--- a/test/functional/p2p_1p1c_network.py
+++ b/test/functional/p2p_1p1c_network.py
@@ -22,12 +22,11 @@ FEERATE_1SAT_VB = Decimal("10")
 
 
 class PackageRelayTest(BitcoinTestFramework):
-    def set_test_params(self):
-        self.setup_clean_chain = True
-        self.num_nodes = 4
-        self.noban_tx_relay = True
-        self.extra_args = [["-acceptnonstdtxn=1", "-maxmempool=5"]] * self.num_nodes
-        self.supports_cli = False
+    setup_clean_chain = True
+    num_nodes = 4
+    noban_tx_relay = True
+    extra_args = [["-acceptnonstdtxn=1", "-maxmempool=5"]] * num_nodes
+    supports_cli = False
 
     def raise_network_minfee(self):
         filler_wallet = MiniWallet(self.nodes[0])
diff --git a/test/functional/p2p_add_connections.py b/test/functional/p2p_add_connections.py
index 2fac166c55..7db92b0219 100644
--- a/test/functional/p2p_add_connections.py
+++ b/test/functional/p2p_add_connections.py
@@ -29,15 +29,14 @@ class P2PFeelerReceiver(P2PInterface):
 
 
 class P2PAddConnections(BitcoinTestFramework):
-    def set_test_params(self):
-        self.num_nodes = 2
-        self.extra_args = [
-            [
-                "-avaproofstakeutxoconfirmations=1",
-                f"-maxavalancheoutbound={MAX_AVALANCHE_OUTBOUND_CONNECTIONS}",
-            ],
-            [],
-        ]
+    num_nodes = 2
+    extra_args = [
+        [
+            "-avaproofstakeutxoconfirmations=1",
+            f"-maxavalancheoutbound={MAX_AVALANCHE_OUTBOUND_CONNECTIONS}",
+        ],
+        [],
+    ]
 
     def setup_network(self):
         self.setup_nodes()
diff --git a/test/functional/p2p_addr_relay.py b/test/functional/p2p_addr_relay.py
index fd6c018b39..3b4a8766fa 100644
--- a/test/functional/p2p_addr_relay.py
+++ b/test/functional/p2p_addr_relay.py
@@ -88,9 +88,8 @@ class AddrTest(BitcoinTestFramework):
     counter = 0
     mocktime = int(time.time())
 
-    def set_test_params(self):
-        self.num_nodes = 1
-        self.extra_args = [["-whitelist=addr@127.0.0.1"]]
+    num_nodes = 1
+    extra_args = [["-whitelist=addr@127.0.0.1"]]
 
     def run_test(self):
         self.oversized_addr_test()
diff --git a/test/functional/p2p_addrfetch.py b/test/functional/p2p_addrfetch.py
index 2d594877c4..e110c51e82 100644
--- a/test/functional/p2p_addrfetch.py
+++ b/test/functional/p2p_addrfetch.py
@@ -20,9 +20,8 @@ ADDR.port = 18444
 
 
 class P2PAddrFetch(BitcoinTestFramework):
-    def set_test_params(self):
-        self.setup_clean_chain = True
-        self.num_nodes = 1
+    setup_clean_chain = True
+    num_nodes = 1
 
     def assert_getpeerinfo(self, *, peer_ids):
         num_peers = len(peer_ids)
diff --git a/test/functional/p2p_addrv2_relay.py b/test/functional/p2p_addrv2_relay.py
index 6ba1a33490..0b6b18b002 100644
--- a/test/functional/p2p_addrv2_relay.py
+++ b/test/functional/p2p_addrv2_relay.py
@@ -38,10 +38,9 @@ class AddrReceiver(P2PInterface):
 
 
 class AddrTest(BitcoinTestFramework):
-    def set_test_params(self):
-        self.setup_clean_chain = True
-        self.num_nodes = 1
-        self.extra_args = [["-whitelist=addr@127.0.0.1"]]
+    setup_clean_chain = True
+    num_nodes = 1
+    extra_args = [["-whitelist=addr@127.0.0.1"]]
 
     def run_test(self):
         self.log.info("Create connection that sends addrv2 messages")
diff --git a/test/functional/p2p_block_sync.py b/test/functional/p2p_block_sync.py
index 853f6be88b..40432d9dd2 100644
--- a/test/functional/p2p_block_sync.py
+++ b/test/functional/p2p_block_sync.py
@@ -11,9 +11,8 @@ from test_framework.test_framework import BitcoinTestFramework
 
 
 class BlockSyncTest(BitcoinTestFramework):
-    def set_test_params(self):
-        self.setup_clean_chain = True
-        self.num_nodes = 3
+    setup_clean_chain = True
+    num_nodes = 3
 
     def setup_network(self):
         self.setup_nodes()
diff --git a/test/functional/p2p_blockfilters.py b/test/functional/p2p_blockfilters.py
index 035f283289..ee2a2eea39 100644
--- a/test/functional/p2p_blockfilters.py
+++ b/test/functional/p2p_blockfilters.py
@@ -39,14 +39,13 @@ class FiltersClient(P2PInterface):
 
 
 class CompactFiltersTest(BitcoinTestFramework):
-    def set_test_params(self):
-        self.setup_clean_chain = True
-        self.rpc_timeout = 480
-        self.num_nodes = 2
-        self.extra_args = [
-            ["-blockfilterindex", "-peerblockfilters"],
-            ["-blockfilterindex"],
-        ]
+    setup_clean_chain = True
+    rpc_timeout = 480
+    num_nodes = 2
+    extra_args = [
+        ["-blockfilterindex", "-peerblockfilters"],
+        ["-blockfilterindex"],
+    ]
 
     def run_test(self):
         # Node 0 supports COMPACT_FILTERS, node 1 does not.
diff --git a/test/functional/p2p_blocksonly.py b/test/functional/p2p_blocksonly.py
index 65319840be..b2dfe967e6 100644
--- a/test/functional/p2p_blocksonly.py
+++ b/test/functional/p2p_blocksonly.py
@@ -13,9 +13,8 @@ from test_framework.wallet import MiniWallet
 
 
 class P2PBlocksOnly(BitcoinTestFramework):
-    def set_test_params(self):
-        self.num_nodes = 1
-        self.extra_args = [["-blocksonly"]]
+    num_nodes = 1
+    extra_args = [["-blocksonly"]]
 
     def run_test(self):
         self.miniwallet = MiniWallet(self.nodes[0])
diff --git a/test/functional/p2p_compactblocks.py b/test/functional/p2p_compactblocks.py
index 2ad0dea39d..1eb2d351d9 100644
--- a/test/functional/p2p_compactblocks.py
+++ b/test/functional/p2p_compactblocks.py
@@ -133,11 +133,10 @@ class TestP2PConn(P2PInterface):
 
 
 class CompactBlocksTest(BitcoinTestFramework):
-    def set_test_params(self):
-        self.setup_clean_chain = True
-        self.num_nodes = 1
-        self.extra_args = [["-acceptnonstdtxn=1"]]
-        self.utxos = []
+    setup_clean_chain = True
+    num_nodes = 1
+    extra_args = [["-acceptnonstdtxn=1"]]
+    utxos = []
 
     def skip_test_if_missing_module(self):
         self.skip_if_no_wallet()
diff --git a/test/functional/p2p_compactblocks_blocksonly.py b/test/functional/p2p_compactblocks_blocksonly.py
index 0cd20b002e..6534e8d6b9 100644
--- a/test/functional/p2p_compactblocks_blocksonly.py
+++ b/test/functional/p2p_compactblocks_blocksonly.py
@@ -21,9 +21,8 @@ from test_framework.util import assert_equal
 
 
 class P2PCompactBlocksBlocksOnly(BitcoinTestFramework):
-    def set_test_params(self):
-        self.extra_args = [["-blocksonly"], [], [], []]
-        self.num_nodes = 4
+    extra_args = [["-blocksonly"], [], [], []]
+    num_nodes = 4
 
     def setup_network(self):
         self.setup_nodes()
diff --git a/test/functional/p2p_compactblocks_hb.py b/test/functional/p2p_compactblocks_hb.py
index 5ef8d7e51c..fdb69e330a 100644
--- a/test/functional/p2p_compactblocks_hb.py
+++ b/test/functional/p2p_compactblocks_hb.py
@@ -10,9 +10,8 @@ from test_framework.util import assert_equal
 class CompactBlocksConnectionTest(BitcoinTestFramework):
     """Test class for verifying selection of HB peer connections."""
 
-    def set_test_params(self):
-        self.setup_clean_chain = True
-        self.num_nodes = 6
+    setup_clean_chain = True
+    num_nodes = 6
 
     def peer_info(self, from_node, to_node):
         """Query from_node for its getpeerinfo about to_node."""
diff --git a/test/functional/p2p_disconnect_ban.py b/test/functional/p2p_disconnect_ban.py
index dced4863f7..9dea34f8df 100644
--- a/test/functional/p2p_disconnect_ban.py
+++ b/test/functional/p2p_disconnect_ban.py
@@ -10,9 +10,8 @@ from test_framework.util import assert_equal, assert_raises_rpc_error
 
 
 class DisconnectBanTest(BitcoinTestFramework):
-    def set_test_params(self):
-        self.num_nodes = 2
-        self.supports_cli = False
+    num_nodes = 2
+    supports_cli = False
 
     def run_test(self):
         self.log.info("Connect nodes both way")
diff --git a/test/functional/p2p_dos_header_tree.py b/test/functional/p2p_dos_header_tree.py
index 5fbadac33f..8f01adeebf 100644
--- a/test/functional/p2p_dos_header_tree.py
+++ b/test/functional/p2p_dos_header_tree.py
@@ -11,11 +11,10 @@ from test_framework.test_framework import BitcoinTestFramework
 
 
 class RejectLowDifficultyHeadersTest(BitcoinTestFramework):
-    def set_test_params(self):
-        self.setup_clean_chain = True
-        self.chain = "testnet3"  # Use testnet chain because it has an early checkpoint
-        self.num_nodes = 2
-        self.extra_args = [["-minimumchainwork=0x0"], ["-minimumchainwork=0x0"]]
+    setup_clean_chain = True
+    chain = "testnet3"  # Use testnet chain because it has an early checkpoint
+    num_nodes = 2
+    extra_args = [["-minimumchainwork=0x0"], ["-minimumchainwork=0x0"]]
 
     def add_options(self, parser):
         parser.add_argument(
diff --git a/test/functional/p2p_eviction.py b/test/functional/p2p_eviction.py
index b1a15098cf..7af87add51 100644
--- a/test/functional/p2p_eviction.py
+++ b/test/functional/p2p_eviction.py
@@ -55,23 +55,22 @@ class SlowAvaP2PInterface(AvaP2PInterface):
 
 
 class P2PEvict(BitcoinTestFramework):
-    def set_test_params(self):
-        self.num_nodes = 1
-        # The choice of maxconnections=188 results in a maximum of 153 inbound
-        # connections (188 - 34 outbound - 1 feeler). The 34 outbounds count is
-        # from 16 full-relay + 16 avalanche + 2 block-only-relay.
-        # 152 inbound peers are protected from eviction:
-        # 4 by netgroup, 4 that sent us blocks, 4 that sent us proofs, 4 that
-        # sent us transactions, 8 via lowest ping time, 128 with the best
-        # avalanche availability score
-        self.extra_args = [
-            [
-                "-maxconnections=188",
-                "-avaproofstakeutxodustthreshold=1000000",
-                "-avaproofstakeutxoconfirmations=1",
-                "-maxavalancheoutbound=16",
-            ]
+    num_nodes = 1
+    # The choice of maxconnections=188 results in a maximum of 153 inbound
+    # connections (188 - 34 outbound - 1 feeler). The 34 outbounds count is
+    # from 16 full-relay + 16 avalanche + 2 block-only-relay.
+    # 152 inbound peers are protected from eviction:
+    # 4 by netgroup, 4 that sent us blocks, 4 that sent us proofs, 4 that
+    # sent us transactions, 8 via lowest ping time, 128 with the best
+    # avalanche availability score
+    extra_args = [
+        [
+            "-maxconnections=188",
+            "-avaproofstakeutxodustthreshold=1000000",
+            "-avaproofstakeutxoconfirmations=1",
+            "-maxavalancheoutbound=16",
         ]
+    ]
 
     def run_test(self):
         # peers that we expect to be protected from eviction
diff --git a/test/functional/p2p_feefilter.py b/test/functional/p2p_feefilter.py
index 8c1f272437..ce9768b810 100644
--- a/test/functional/p2p_feefilter.py
+++ b/test/functional/p2p_feefilter.py
@@ -43,20 +43,19 @@ class TestP2PConn(P2PInterface):
 
 
 class FeeFilterTest(BitcoinTestFramework):
-    def set_test_params(self):
-        self.num_nodes = 2
-        self.noban_tx_relay = True
-        # We lower the various required feerates for this test
-        # to catch a corner-case where feefilter used to slightly undercut
-        # mempool and wallet feerate calculation based on GetFee
-        # rounding down 3 places, leading to stranded transactions.
-        # See issue #16499
-        self.extra_args = [
-            [
-                "-minrelaytxfee=1",
-                "-mintxfee=1",
-            ]
-        ] * self.num_nodes
+    num_nodes = 2
+    noban_tx_relay = True
+    # We lower the various required feerates for this test
+    # to catch a corner-case where feefilter used to slightly undercut
+    # mempool and wallet feerate calculation based on GetFee
+    # rounding down 3 places, leading to stranded transactions.
+    # See issue #16499
+    extra_args = [
+        [
+            "-minrelaytxfee=1",
+            "-mintxfee=1",
+        ]
+    ] * num_nodes
 
     def run_test(self):
         self.test_feefilter_forcerelay()
diff --git a/test/functional/p2p_filter.py b/test/functional/p2p_filter.py
index 09b915ea02..a50fd61fd3 100644
--- a/test/functional/p2p_filter.py
+++ b/test/functional/p2p_filter.py
@@ -107,10 +107,9 @@ class P2PBloomFilter(P2PInterface):
 
 
 class FilterTest(BitcoinTestFramework):
-    def set_test_params(self):
-        self.num_nodes = 1
-        self.noban_tx_relay = True
-        self.extra_args = [["-peerbloomfilters"]]
+    num_nodes = 1
+    noban_tx_relay = True
+    extra_args = [["-peerbloomfilters"]]
 
     def generatetoscriptpubkey(self, scriptpubkey):
         """Helper to generate a single block to the given scriptPubKey."""
diff --git a/test/functional/p2p_fingerprint.py b/test/functional/p2p_fingerprint.py
index 37ab9170f3..6596565ee4 100644
--- a/test/functional/p2p_fingerprint.py
+++ b/test/functional/p2p_fingerprint.py
@@ -24,9 +24,8 @@ from test_framework.util import assert_equal
 
 
 class P2PFingerprintTest(BitcoinTestFramework):
-    def set_test_params(self):
-        self.setup_clean_chain = True
-        self.num_nodes = 1
+    setup_clean_chain = True
+    num_nodes = 1
 
     # Build a chain of blocks on top of given one
     def build_chain(self, nblocks, prev_hash, prev_height, prev_median_time):
diff --git a/test/functional/p2p_getaddr_caching.py b/test/functional/p2p_getaddr_caching.py
index 5ba0e01a84..7f8d9bddfa 100644
--- a/test/functional/p2p_getaddr_caching.py
+++ b/test/functional/p2p_getaddr_caching.py
@@ -34,11 +34,10 @@ class AddrReceiver(P2PInterface):
 
 
 class AddrTest(BitcoinTestFramework):
-    def set_test_params(self):
-        self.num_nodes = 1
-        # TODO When Backporting core#25096, please also includes changes from:
-        #  - core#25312
-        #  - core#25333
+    num_nodes = 1
+    # TODO When Backporting core#25096, please also includes changes from:
+    #  - core#25312
+    #  - core#25333
 
     def run_test(self):
         self.log.info("Fill peer AddrMan with a lot of records")
diff --git a/test/functional/p2p_getdata.py b/test/functional/p2p_getdata.py
index e07bfdd58c..94410deefb 100644
--- a/test/functional/p2p_getdata.py
+++ b/test/functional/p2p_getdata.py
@@ -20,8 +20,7 @@ class P2PStoreBlock(P2PInterface):
 
 
 class GetdataTest(BitcoinTestFramework):
-    def set_test_params(self):
-        self.num_nodes = 1
+    num_nodes = 1
 
     def run_test(self):
         p2p_block_store = self.nodes[0].add_p2p_connection(P2PStoreBlock())
diff --git a/test/functional/p2p_i2p_ports.py b/test/functional/p2p_i2p_ports.py
index 2d55c93628..82ca861858 100644
--- a/test/functional/p2p_i2p_ports.py
+++ b/test/functional/p2p_i2p_ports.py
@@ -11,10 +11,9 @@ from test_framework.test_framework import BitcoinTestFramework
 
 
 class I2PPorts(BitcoinTestFramework):
-    def set_test_params(self):
-        self.num_nodes = 1
-        # The test assumes that an I2P SAM proxy is not listening here.
-        self.extra_args = [["-i2psam=127.0.0.1:60000"]]
+    num_nodes = 1
+    # The test assumes that an I2P SAM proxy is not listening here.
+    extra_args = [["-i2psam=127.0.0.1:60000"]]
 
     def run_test(self):
         node = self.nodes[0]
diff --git a/test/functional/p2p_i2p_sessions.py b/test/functional/p2p_i2p_sessions.py
index ab348dc40e..b89469801c 100644
--- a/test/functional/p2p_i2p_sessions.py
+++ b/test/functional/p2p_i2p_sessions.py
@@ -9,13 +9,12 @@ from test_framework.test_framework import BitcoinTestFramework
 
 
 class I2PSessions(BitcoinTestFramework):
-    def set_test_params(self):
-        self.num_nodes = 2
-        # The test assumes that an I2P SAM proxy is not listening here.
-        self.extra_args = [
-            ["-i2psam=127.0.0.1:60000", "-i2pacceptincoming=1"],
-            ["-i2psam=127.0.0.1:60000", "-i2pacceptincoming=0"],
-        ]
+    num_nodes = 2
+    # The test assumes that an I2P SAM proxy is not listening here.
+    extra_args = [
+        ["-i2psam=127.0.0.1:60000", "-i2pacceptincoming=1"],
+        ["-i2psam=127.0.0.1:60000", "-i2pacceptincoming=0"],
+    ]
 
     def run_test(self):
         addr = "zsxwyo6qcn3chqzwxnseusqgsnuw3maqnztkiypyfxtya4snkoka.b32.i2p"
diff --git a/test/functional/p2p_ibd_stalling.py b/test/functional/p2p_ibd_stalling.py
index bc111ba4e4..9ecee162f1 100644
--- a/test/functional/p2p_ibd_stalling.py
+++ b/test/functional/p2p_ibd_stalling.py
@@ -37,9 +37,8 @@ class P2PStaller(P2PDataStore):
 
 
 class P2PIBDStallingTest(BitcoinTestFramework):
-    def set_test_params(self):
-        self.setup_clean_chain = True
-        self.num_nodes = 1
+    setup_clean_chain = True
+    num_nodes = 1
 
     def run_test(self):
         NUM_BLOCKS = 1025
diff --git a/test/functional/p2p_ibd_txrelay.py b/test/functional/p2p_ibd_txrelay.py
index a671544688..800320e6d4 100644
--- a/test/functional/p2p_ibd_txrelay.py
+++ b/test/functional/p2p_ibd_txrelay.py
@@ -32,13 +32,12 @@ NORMAL_FEE_FILTER = Decimal(100) / XEC
 
 
 class P2PIBDTxRelayTest(BitcoinTestFramework):
-    def set_test_params(self):
-        self.setup_clean_chain = True
-        self.num_nodes = 2
-        self.extra_args = [
-            [f"-minrelaytxfee={NORMAL_FEE_FILTER}"],
-            [f"-minrelaytxfee={NORMAL_FEE_FILTER}"],
-        ]
+    setup_clean_chain = True
+    num_nodes = 2
+    extra_args = [
+        [f"-minrelaytxfee={NORMAL_FEE_FILTER}"],
+        [f"-minrelaytxfee={NORMAL_FEE_FILTER}"],
+    ]
 
     def run_test(self):
         self.log.info("Check that nodes set minfilter to MAX_MONEY while still in IBD")
diff --git a/test/functional/p2p_initial_headers_sync.py b/test/functional/p2p_initial_headers_sync.py
index 0ccc1dd0f4..6ebf020c8a 100644
--- a/test/functional/p2p_initial_headers_sync.py
+++ b/test/functional/p2p_initial_headers_sync.py
@@ -17,9 +17,8 @@ from test_framework.util import assert_equal
 
 
 class HeadersSyncTest(BitcoinTestFramework):
-    def set_test_params(self):
-        self.setup_clean_chain = True
-        self.num_nodes = 1
+    setup_clean_chain = True
+    num_nodes = 1
 
     def announce_random_block(self, peers):
         new_block_announcement = msg_inv(
diff --git a/test/functional/p2p_inv_download.py b/test/functional/p2p_inv_download.py
index 8258eca827..cadada9f21 100644
--- a/test/functional/p2p_inv_download.py
+++ b/test/functional/p2p_inv_download.py
@@ -128,15 +128,14 @@ def skip(context):
 
 
 class InventoryDownloadTest(BitcoinTestFramework):
-    def set_test_params(self):
-        self.num_nodes = 2
-        self.extra_args = [
-            [
-                "-avaproofstakeutxodustthreshold=1000000",
-                "-avaproofstakeutxoconfirmations=1",
-                "-avacooldown=0",
-            ]
-        ] * self.num_nodes
+    num_nodes = 2
+    extra_args = [
+        [
+            "-avaproofstakeutxodustthreshold=1000000",
+            "-avaproofstakeutxoconfirmations=1",
+            "-avacooldown=0",
+        ]
+    ] * num_nodes
 
     def test_data_requests(self, context):
         self.log.info("Test that we request data from all our peers, eventually")
diff --git a/test/functional/p2p_invalid_block.py b/test/functional/p2p_invalid_block.py
index ef83fc3c3e..5566277d76 100644
--- a/test/functional/p2p_invalid_block.py
+++ b/test/functional/p2p_invalid_block.py
@@ -31,10 +31,9 @@ from test_framework.util import assert_equal
 
 
 class InvalidBlockRequestTest(BitcoinTestFramework):
-    def set_test_params(self):
-        self.num_nodes = 1
-        self.setup_clean_chain = True
-        self.noban_tx_relay = True
+    num_nodes = 1
+    setup_clean_chain = True
+    noban_tx_relay = True
 
     def run_test(self):
         # Add p2p connection to node0
diff --git a/test/functional/p2p_invalid_locator.py b/test/functional/p2p_invalid_locator.py
index cabeb87ff1..3239255ba6 100644
--- a/test/functional/p2p_invalid_locator.py
+++ b/test/functional/p2p_invalid_locator.py
@@ -9,8 +9,7 @@ from test_framework.test_framework import BitcoinTestFramework
 
 
 class InvalidLocatorTest(BitcoinTestFramework):
-    def set_test_params(self):
-        self.num_nodes = 1
+    num_nodes = 1
 
     def run_test(self):
         # Convenience reference to the node
diff --git a/test/functional/p2p_invalid_messages.py b/test/functional/p2p_invalid_messages.py
index 8031e6bc2d..5153dd3b25 100644
--- a/test/functional/p2p_invalid_messages.py
+++ b/test/functional/p2p_invalid_messages.py
@@ -55,10 +55,9 @@ class SenderOfAddrV2(P2PInterface):
 
 
 class InvalidMessagesTest(BitcoinTestFramework):
-    def set_test_params(self):
-        self.num_nodes = 1
-        self.setup_clean_chain = True
-        self.extra_args = [["-whitelist=addr@127.0.0.1"]]
+    num_nodes = 1
+    setup_clean_chain = True
+    extra_args = [["-whitelist=addr@127.0.0.1"]]
 
     def run_test(self):
         self.test_buffer()
diff --git a/test/functional/p2p_invalid_tx.py b/test/functional/p2p_invalid_tx.py
index 0441a8e7a2..025b1b6d13 100644
--- a/test/functional/p2p_invalid_tx.py
+++ b/test/functional/p2p_invalid_tx.py
@@ -22,14 +22,13 @@ from test_framework.util import assert_equal
 
 
 class InvalidTxRequestTest(BitcoinTestFramework):
-    def set_test_params(self):
-        self.num_nodes = 1
-        self.extra_args = [
-            [
-                "-acceptnonstdtxn=1",
-            ]
+    num_nodes = 1
+    extra_args = [
+        [
+            "-acceptnonstdtxn=1",
         ]
-        self.setup_clean_chain = True
+    ]
+    setup_clean_chain = True
 
     def bootstrap_p2p(self, *, num_connections=1):
         """Add a P2P connection to the node.
diff --git a/test/functional/p2p_leak.py b/test/functional/p2p_leak.py
index 970bc49511..8be7828443 100644
--- a/test/functional/p2p_leak.py
+++ b/test/functional/p2p_leak.py
@@ -129,8 +129,7 @@ class P2PVersionStore(P2PInterface):
 
 
 class P2PLeakTest(BitcoinTestFramework):
-    def set_test_params(self):
-        self.num_nodes = 1
+    num_nodes = 1
 
     def run_test(self):
         # Peer that never sends a version. We will send a bunch of messages
diff --git a/test/functional/p2p_leak_tx.py b/test/functional/p2p_leak_tx.py
index 11d0a2e8e5..6abb99da74 100644
--- a/test/functional/p2p_leak_tx.py
+++ b/test/functional/p2p_leak_tx.py
@@ -21,19 +21,18 @@ class P2PNode(P2PDataStore):
 
 
 class P2PLeakTxTest(BitcoinTestFramework):
-    def set_test_params(self):
-        self.num_nodes = 1
-        self.noban_tx_relay = True
-        self.extra_args = [
-            [
-                "-avacooldown=0",
-                "-avaproofstakeutxoconfirmations=1",
-                # Low enough for coinbase transactions to be staked in valid proofs
-                "-avaproofstakeutxodustthreshold=1000000",
-                "-avaminquorumstake=0",
-                "-avaminavaproofsnodecount=0",
-            ],
-        ]
+    num_nodes = 1
+    noban_tx_relay = True
+    extra_args = [
+        [
+            "-avacooldown=0",
+            "-avaproofstakeutxoconfirmations=1",
+            # Low enough for coinbase transactions to be staked in valid proofs
+            "-avaproofstakeutxodustthreshold=1000000",
+            "-avaminquorumstake=0",
+            "-avaminavaproofsnodecount=0",
+        ],
+    ]
 
     def run_test(self):
         # The block and tx generating node
diff --git a/test/functional/p2p_message_capture.py b/test/functional/p2p_message_capture.py
index fdceb0e5c0..f9a9a7d74f 100644
--- a/test/functional/p2p_message_capture.py
+++ b/test/functional/p2p_message_capture.py
@@ -57,10 +57,9 @@ def mini_parser(dat_file):
 
 
 class MessageCaptureTest(BitcoinTestFramework):
-    def set_test_params(self):
-        self.num_nodes = 1
-        self.extra_args = [["-capturemessages"]]
-        self.setup_clean_chain = True
+    num_nodes = 1
+    extra_args = [["-capturemessages"]]
+    setup_clean_chain = True
 
     def run_test(self):
         capturedir = os.path.join(self.nodes[0].datadir, "regtest/message_capture")
diff --git a/test/functional/p2p_mutated_blocks.py b/test/functional/p2p_mutated_blocks.py
index bd9fb19d2f..617d21c2ba 100644
--- a/test/functional/p2p_mutated_blocks.py
+++ b/test/functional/p2p_mutated_blocks.py
@@ -24,9 +24,8 @@ from test_framework.wallet import MiniWallet
 
 
 class MutatedBlocksTest(BitcoinTestFramework):
-    def set_test_params(self):
-        self.setup_clean_chain = True
-        self.num_nodes = 1
+    setup_clean_chain = True
+    num_nodes = 1
 
     def run_test(self):
         self.wallet = MiniWallet(self.nodes[0])
diff --git a/test/functional/p2p_net_deadlock.py b/test/functional/p2p_net_deadlock.py
index f917f40b79..1e6059e661 100644
--- a/test/functional/p2p_net_deadlock.py
+++ b/test/functional/p2p_net_deadlock.py
@@ -19,8 +19,7 @@ from test_framework.wallet import MiniWallet
 
 
 class NetDeadlockTest(BitcoinTestFramework):
-    def set_test_params(self):
-        self.num_nodes = 2
+    num_nodes = 2
 
     def run_test(self):
         node0 = self.nodes[0]
diff --git a/test/functional/p2p_nobloomfilter_messages.py b/test/functional/p2p_nobloomfilter_messages.py
index 4966bf115f..780d29d340 100644
--- a/test/functional/p2p_nobloomfilter_messages.py
+++ b/test/functional/p2p_nobloomfilter_messages.py
@@ -22,10 +22,9 @@ from test_framework.util import assert_equal
 
 
 class P2PNoBloomFilterMessages(BitcoinTestFramework):
-    def set_test_params(self):
-        self.setup_clean_chain = True
-        self.num_nodes = 1
-        self.extra_args = [["-peerbloomfilters=0"]]
+    setup_clean_chain = True
+    num_nodes = 1
+    extra_args = [["-peerbloomfilters=0"]]
 
     def test_message_causes_disconnect(self, message):
         """Add a p2p connection that sends a message and check that it disconnects."""
diff --git a/test/functional/p2p_node_network_limited.py b/test/functional/p2p_node_network_limited.py
index a9447c90ea..fd57a23b81 100644
--- a/test/functional/p2p_node_network_limited.py
+++ b/test/functional/p2p_node_network_limited.py
@@ -44,10 +44,9 @@ class P2PIgnoreInv(P2PInterface):
 
 
 class NodeNetworkLimitedTest(BitcoinTestFramework):
-    def set_test_params(self):
-        self.setup_clean_chain = True
-        self.num_nodes = 3
-        self.extra_args = [["-prune=550", "-addrmantest"], [], []]
+    setup_clean_chain = True
+    num_nodes = 3
+    extra_args = [["-prune=550", "-addrmantest"], [], []]
 
     def disconnect_all(self):
         self.disconnect_nodes(0, 1)
diff --git a/test/functional/p2p_opportunistic_1p1c.py b/test/functional/p2p_opportunistic_1p1c.py
index cc6b5c5ec9..bd542de98e 100644
--- a/test/functional/p2p_opportunistic_1p1c.py
+++ b/test/functional/p2p_opportunistic_1p1c.py
@@ -45,16 +45,15 @@ def cleanup(func):
 
 
 class PackageRelayTest(BitcoinTestFramework):
-    def set_test_params(self):
-        self.setup_clean_chain = True
-        self.num_nodes = 1
-        self.extra_args = [
-            [
-                "-acceptnonstdtxn=1",
-                "-maxmempool=5",
-            ]
+    setup_clean_chain = True
+    num_nodes = 1
+    extra_args = [
+        [
+            "-acceptnonstdtxn=1",
+            "-maxmempool=5",
         ]
-        self.supports_cli = False
+    ]
+    supports_cli = False
 
     def create_tx_below_mempoolminfee(self, wallet):
         """Create a 1-input 1sat/vB transaction using a confirmed UTXO.
diff --git a/test/functional/p2p_orphan_handling.py b/test/functional/p2p_orphan_handling.py
index 3fe8b363fb..b03cadc55d 100644
--- a/test/functional/p2p_orphan_handling.py
+++ b/test/functional/p2p_orphan_handling.py
@@ -109,9 +109,8 @@ class PeerTxRelayer(P2PTxInvStore):
 
 
 class OrphanHandlingTest(BitcoinTestFramework):
-    def set_test_params(self):
-        self.num_nodes = 1
-        self.extra_args = [[]]
+    num_nodes = 1
+    extra_args = [[]]
 
     def create_parent_and_child(self):
         """Create package with 1 parent and 1 child, normal fees (no cpfp)."""
diff --git a/test/functional/p2p_permissions.py b/test/functional/p2p_permissions.py
index b9bb3e6b6d..503fb3ad69 100644
--- a/test/functional/p2p_permissions.py
+++ b/test/functional/p2p_permissions.py
@@ -17,9 +17,8 @@ from test_framework.util import append_config, assert_equal, p2p_port, tor_port
 
 
 class P2PPermissionsTests(BitcoinTestFramework):
-    def set_test_params(self):
-        self.num_nodes = 2
-        self.setup_clean_chain = True
+    num_nodes = 2
+    setup_clean_chain = True
 
     def run_test(self):
         self.check_tx_relay()
diff --git a/test/functional/p2p_ping.py b/test/functional/p2p_ping.py
index bfdd446e86..d4bcf55c78 100644
--- a/test/functional/p2p_ping.py
+++ b/test/functional/p2p_ping.py
@@ -25,12 +25,11 @@ class NodeNoPong(P2PInterface):
 
 
 class PingPongTest(BitcoinTestFramework):
-    def set_test_params(self):
-        self.setup_clean_chain = True
-        self.num_nodes = 1
-        # Set the peer connection timeout low. It does not matter for this
-        # test, as long as it is less than TIMEOUT_INTERVAL.
-        self.extra_args = [["-peertimeout=1"]]
+    setup_clean_chain = True
+    num_nodes = 1
+    # Set the peer connection timeout low. It does not matter for this
+    # test, as long as it is less than TIMEOUT_INTERVAL.
+    extra_args = [["-peertimeout=1"]]
 
     def check_peer_info(self, *, pingtime, minping, pingwait):
         stats = self.nodes[0].getpeerinfo()[0]
diff --git a/test/functional/p2p_sendheaders.py b/test/functional/p2p_sendheaders.py
index 6eb463ea9e..550f631dfb 100644
--- a/test/functional/p2p_sendheaders.py
+++ b/test/functional/p2p_sendheaders.py
@@ -193,10 +193,9 @@ class BaseNode(P2PInterface):
 
 
 class SendHeadersTest(BitcoinTestFramework):
-    def set_test_params(self):
-        self.setup_clean_chain = True
-        self.num_nodes = 2
-        self.extra_args = [["-noparkdeepreorg"], ["-noparkdeepreorg"]]
+    setup_clean_chain = True
+    num_nodes = 2
+    extra_args = [["-noparkdeepreorg"], ["-noparkdeepreorg"]]
 
     def mine_blocks(self, count):
         """Mine count blocks and return the new tip."""
diff --git a/test/functional/p2p_timeouts.py b/test/functional/p2p_timeouts.py
index 5c02484b77..93e3456a21 100644
--- a/test/functional/p2p_timeouts.py
+++ b/test/functional/p2p_timeouts.py
@@ -30,11 +30,10 @@ class TestP2PConn(P2PInterface):
 
 
 class TimeoutsTest(BitcoinTestFramework):
-    def set_test_params(self):
-        self.setup_clean_chain = True
-        self.num_nodes = 1
-        # set timeout to receive version/verack to 3 seconds
-        self.extra_args = [["-peertimeout=3"]]
+    setup_clean_chain = True
+    num_nodes = 1
+    # set timeout to receive version/verack to 3 seconds
+    extra_args = [["-peertimeout=3"]]
 
     def mock_forward(self, delta):
         self.mock_time += delta
diff --git a/test/functional/p2p_tx_privacy.py b/test/functional/p2p_tx_privacy.py
index 5bf7310b05..2699048690 100644
--- a/test/functional/p2p_tx_privacy.py
+++ b/test/functional/p2p_tx_privacy.py
@@ -44,8 +44,7 @@ class P2PTxSpy(P2PInterface):
 
 
 class TxPrivacyTest(BitcoinTestFramework):
-    def set_test_params(self):
-        self.num_nodes = 1
+    num_nodes = 1
 
     def run_test(self):
         self.wallet = MiniWallet(self.nodes[0])
diff --git a/test/functional/p2p_unrequested_blocks.py b/test/functional/p2p_unrequested_blocks.py
index 07fa3c9ab4..e1b51a4cd0 100644
--- a/test/functional/p2p_unrequested_blocks.py
+++ b/test/functional/p2p_unrequested_blocks.py
@@ -71,10 +71,9 @@ from test_framework.util import assert_equal, assert_raises_rpc_error
 
 
 class AcceptBlockTest(BitcoinTestFramework):
-    def set_test_params(self):
-        self.setup_clean_chain = True
-        self.num_nodes = 2
-        self.extra_args = [["-noparkdeepreorg"], ["-minimumchainwork=0x10"]]
+    setup_clean_chain = True
+    num_nodes = 2
+    extra_args = [["-noparkdeepreorg"], ["-minimumchainwork=0x10"]]
 
     def setup_network(self):
         self.setup_nodes()
-- 
2.50.1 (Apple Git-155)

